src/gflownet/tasks/config.py

```diff
@@ -0,0 +1,91 @@
+from dataclasses import dataclass, field
+from typing import Any, List, Optional, Tuple
+
+
+@dataclass
+class SEHTaskConfig:
+    """Config for the SEHTask
+
+    Attributes
+    ----------
+
+    temperature_sample_dist : str
+        The distribution to sample the inverse temperature from. Can be one of:
+        - "uniform": uniform distribution
+        - "loguniform": log-uniform distribution
+        - "gamma": gamma distribution
+        - "constant": constant temperature
+    temperature_dist_params : List[Any]
+        The parameters of the temperature distribution. E.g. for the "uniform" distribution, this is the range.
+    num_thermometer_dim : int
+        The number of thermometer encoding dimensions to use.
+    """
+
+    # TODO: a proper class for temperature-conditional sampling
+    temperature_sample_dist: str = "uniform"
```

Comment:
If not any string is allowed you can have something like `Literal["uniform", "method2", "method3"]` (or enum).